### PR TITLE
Move state access methods to DB

### PR DESCRIPTION
**Pull request by platfowner**
Change summary:
- Use db's methods to get/set account nonce and timestamp
- Move getValueWithStateVersion() from node to DB (getValueFromStateRoot())
- Move getAppStakesTotal() from TransactionPool to DB
- Add cloneToTempVersion() and simplify temp db creation logic
- Log stack trace for errors
- Remove some unused methods
- Log stack trace in exceptional cases.

Related issue: https://github.com/ainblockchain/ain-blockchain/issues/388

### CODE REVIEW

File: db/index.js [hunk 1]
```diff
@@ -262,7 +262,8 @@ class DB {
   /**
    * Returns reference to the input path for reading if exists, otherwise null.
    */
-  static getRefForReading(node, fullPath) {
+  static getRefForReading(stateRoot, fullPath) {
```
Comment (liayoo) [inline db/index.js:265]: s/getRefForReading/getRefForReadingFromStateRoot

File: db/index.js [hunk 2]
```diff
@@ -290,8 +291,8 @@ class DB {
   // - Reference from root_a: child_1a -> child_2a -> child_3a
   // - Reference from root_b: child_1b -> child_2 -> child_3 (not affected)
   //
-  getRefForWriting(fullPath) {
-    let node = this.stateRoot;
+  static getRefForWriting(stateRoot, fullPath) {
```
Comment (liayoo) [inline db/index.js:294]: s/getRefForWriting/getRefForWritingFromStateRoot

File: db/index.js [hunk 3]
```diff
@@ -537,36 +548,61 @@ class DB {
...
+
+    return true;
+  }
+
+  updateAccountNonceAndTimestamp(address, nonce, timestamp) {
+    return DB.updateAccountNonceAndTimestampToStateRoot(
+        this.stateRoot, this.stateVersion, address, nonce, timestamp);
+  }
+
+  // TODO(platfowner): Use shallow fetch once available.
```
Comment (liayoo) [inline db/index.js:594]: 👍

File: db/index.js [hunk 4]
```diff
@@ -537,36 +548,61 @@ class DB {
...
+  static getAppStakesTotal(stateRoot) {
```
Comment (liayoo) [inline db/index.js:595]: ditto

File: db/index.js [hunk 5]
```diff
@@ -537,36 +548,61 @@ class DB {
...
+    const appStakes = DB.getValueFromStateRoot(stateRoot, PredefinedDbPaths.STAKING) || {};
+    return Object.keys(appStakes).reduce((acc, cur) => {
+      if (cur === PredefinedDbPaths.CONSENSUS) return acc;
+      return acc + _.get(appStakes[cur], PredefinedDbPaths.STAKING_BALANCE_TOTAL, 0);
+    }, 0);
+  }
+
+  static getAppStake(stateRoot, appName) {
```
Comment (liayoo) [inline db/index.js:603]: ditto

File: tx-pool/index.js [hunk 6]
```diff
@@ -2,10 +2,11 @@
 const logger = require('../logger')('TX_POOL');
 const _ = require('lodash');
 const {
+  FeatureFlags,
   TRANSACTION_POOL_TIMEOUT_MS,
   TRANSACTION_TRACKER_TIMEOUT_MS,
   LIGHTWEIGHT,
-	BANDWIDTH_BUDGET_PER_BLOCK,
+  BANDWIDTH_BUDGET_PER_BLOCK,
 	SERVICE_BANDWIDTH_BUDGET_PER_BLOCK,
```
Comment (liayoo) [inline tx-pool/index.js:10]: Could you fix this indentation for me?

File: db/index.js (hunk 1 above)
Comment (platfowner) [reply db/index.js]: Done, thanks.

File: db/index.js (hunk 2 above)
Comment (platfowner) [reply db/index.js]: Done, thanks.

File: db/index.js (hunk 4 above)
Comment (platfowner) [reply db/index.js]: Renamed, thanks.

File: db/index.js (hunk 5 above)
Comment (platfowner) [reply db/index.js]: Renamed, thanks.

File: tx-pool/index.js (hunk 6 above)
Comment (platfowner) [reply tx-pool/index.js]: Fixed.

**Comment by platfowner** [comment]
Comments addressed, PTAL @liayoo

**Comment by liayoo** [comment]
LGTM. Thanks for fixing the issue! 🙏

**Review by liayoo — APPROVED** [review]

**Comment by platfowner** [comment]
Thanks for the review! Now merging..

**Pull request merged**